src/Blocks/BlockPiston.cpp

```diff
@@ -140,12 +156,9 @@ void cBlockPistonHandler::PushBlocks(
 
 
 
-
-bool cBlockPistonHandler::CanPushBlock(
-	const Vector3i & a_BlockPos, cWorld * a_World, bool a_RequirePushable,
-	Vector3iSet & a_BlocksPushed, const Vector3i & a_PushDir
-)
+bool cBlockPistonHandler::CanPushBlock(const Vector3i & a_BlockPos, cWorld * a_World, bool a_RequirePushable, Vector3iSet & a_BlocksPushed, const Vector3i & a_PushDir)
 {
+	bool result = false;
```

Comment:
Optimization. You only have to look in one spot to see where it's returning and avoids duplication of effort.

Also, it's easier to debug because you can more easily follow the code flow.

Functions are guaranteed to return at end of the method. If you need additional logic, you simply check to see if you're interested in the state of the method, then your code runs. Or it doesn't.